doc/en/changelog.rst

```diff
@@ -28,6 +28,420 @@ with advance notice in the **Deprecations** section of releases.
+  The format supports ``or``, ``and``, ``not``, parenthesis and general identifiers to match against.
+  Python constants, keywords or other operators are no longer evaluated differently.
+
+
+- `#7135 <https://github.com/pytest-dev/pytest/issues/7135>`_: Pytest now uses its own ``TerminalWriter`` class instead of using the one from the ``py`` library.
+  Plugins generally access this class through ``TerminalReporter.writer``, ``TerminalReporter.write()``
+  (and similar methods), or ``_pytest.config.create_terminal_writer()``.
+
+  The following breaking changes were made:
+
+  - Output (``write()`` method and others) no longer flush implicitly; the flushing behavior
+    of the underlying file is respected. To flush explicitly (for example, if you
+    want output to be shown before an end-of-line is printed), use ``write(flush=True)`` or
+    ``terminal_writer.flush()``.
+  - Explicit Windows console support was removed, delegated to the colorama library.
+  - Support for writing ``bytes`` was removed.
+  - The ``reline`` method and ``chars_on_current_line`` property were removed.
+  - The ``stringio`` and ``encoding`` arguments was removed.
+  - Support for passing a callable instead of a file was removed.
+
+
+- `#7224 <https://github.com/pytest-dev/pytest/issues/7224>`_: The `item.catch_log_handler` and `item.catch_log_handlers` attributes, set by the
+  logging plugin and never meant to be public , are no longer available.
+
+  The deprecated ``--no-print-logs`` option is removed. Use ``--show-capture`` instead.
+
+
+- `#7226 <https://github.com/pytest-dev/pytest/issues/7226>`_: Removed the unused ``args`` parameter from ``pytest.Function.__init__``.
+
+
+- `#7418 <https://github.com/pytest-dev/pytest/issues/7418>`_: Remove the `pytest_doctest_prepare_content` hook specification. This hook
+  hasn't been triggered by pytest for at least 10 years.
+
+
+- `#7438 <https://github.com/pytest-dev/pytest/issues/7438>`_: Some changes were made to the internal ``_pytest._code.source``, listed here
+  for the benefit of plugin authors who may be using it:
+
+  - The ``deindent`` argument to ``Source()`` has been removed, now it is always true.
+  - Support for zero or multiple arguments to ``Source()`` has been removed.
+  - Support for comparing ``Source`` with an ``str`` has been removed.
+  - The methods ``Source.isparseable()`` and ``Source.putaround()`` have been removed.
+  - The method ``Source.compile()`` and function ``_pytest._code.compile()`` have
+    been removed; use plain ``compile()`` instead.
+  - The function ``_pytest._code.source.getsource()`` has been removed; use
+    ``Source()`` directly instead.
+
+
+
+Deprecations
+------------
+
+- `#6981 <https://github.com/pytest-dev/pytest/issues/6981>`_: Deprecate the ``pytest.collect`` module as it's just aliases into ``pytest``.
```

Comment:
There is a known regression with this already: https://github.com/pytest-dev/pytest/pull/6981#issuecomment-606496652
(IIRC it does not happen with https://github.com/pytest-dev/pytest/pull/6803)